librz/include/rz_util/rz_float.h

```diff
@@ -100,6 +100,31 @@ typedef struct rz_float_t {
 	RzFloatException exception; ///< exception of float operations
 } RzFloat;
 
+static inline bool rz_float_is_neg_zero_f32(float zero) {
+	if (zero == 0.0f) {
+		return 1.0f / zero == -INFINITY;
+	}
+	return false;
+}
+
+static inline bool rz_float_is_neg_zero_f64(double zero) {
+	if (zero == 0.0) {
+		return 1.0 / zero == -INFINITY;
+	}
+	return false;
+}
+
+static inline bool rz_float_is_neg_zero_long_double(long double zero) {
+	if (zero == 0.0l) {
```

Comment:
is this actually correct?

Reply:
Yes, this is likely why we didn't detect it before. Because it compares against `0.0` in the `rz_float_new_from` functions.
And in IEEE 754 `-0.0 == 0.0`.
See https://en.wikipedia.org/wiki/Signed_zero#Comparisons